### PR TITLE
#12 :bug: support array right hand side values with managed dependencies

### PR DESCRIPTION
**Pull request by d-ryan-ashcraft**

### CODE REVIEW

File: habushu-maven-plugin/pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <parent>
         <groupId>org.technologybrewery.habushu</groupId>
         <artifactId>habushu</artifactId>
-        <version>2.5.0-SNAPSHOT</version>
+        <version>2.4.1-SNAPSHOT</version>
```
Comment (kdendtler) [inline habushu-maven-plugin/pom.xml:7]: Q: Was this just reverting an incorrect value?

**Review by jocobtt — APPROVED** [review]

File: habushu-maven-plugin/pom.xml (hunk 1 above)
Comment (d-ryan-ashcraft) [reply habushu-maven-plugin/pom.xml:7]: No, this is a patch vice minor release

Comment (kdendtler) [reply habushu-maven-plugin/pom.xml:7]: Should it be done on the master branch then? If there are no other dev changes then it won't have any functional change, but just verifying.

Comment (d-ryan-ashcraft) [reply habushu-maven-plugin/pom.xml:7]: there's not really a need for a master branch on a library.  2.5.0 was just set as the next dev version.  i'll set it back after this to 2.5.0 again.

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/InstallDependenciesMojo.java [hunk 2]
```diff
@@ -386,6 +388,16 @@ protected void performPendingDependencyReplacements(Map<String, TomlReplacementT
...
     }
 
+    /**
+     * Handles escaping with double quotes only if the value is not an array.
+     *
+     * @param valueToEscape value to potentially escape
+     * @return value ready to write to toml file
+     */
+    protected static String escapeTomlRightHandSide(String valueToEscape) {
+        return (!valueToEscape.contains("{")) ? DOUBLE_QUOTE + valueToEscape + DOUBLE_QUOTE : valueToEscape;
```
Comment (kdendtler) [inline habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/InstallDependenciesMojo.java:398]: Q: For clarity, this isn't actually checking if there is an array but a table instead? 

https://toml.io/en/v1.0.0#inline-table

**Review by epwilkins — APPROVED** [review]

File: habushu-maven-plugin/pom.xml (hunk 1 above)
Comment (kdendtler) [reply habushu-maven-plugin/pom.xml:7]: I'm fine with the management on this versioning scheme but I think there's still a case to be made that we should have a master/main branch specifically for this kind of thing in the future. If we want to release a minor patch quickly but have inflight changes on dev it seems like that would be much easier and safer with separate branches. Can move that to another conversation though.

File: habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/InstallDependenciesMojo.java (hunk 2 above)
Comment (d-ryan-ashcraft) [reply habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/InstallDependenciesMojo.java:398]: I guess that's what I get for commenting a protected method.  Happy to make the change.

File: habushu-maven-plugin/pom.xml (hunk 1 above)
Comment (d-ryan-ashcraft) [reply habushu-maven-plugin/pom.xml:7]: When we have that problem, I'd be happy to discuss.  In the very short run, I think we try to keep things simple.

**Review by kdendtler — APPROVED** [review]

**Pull request merged**